src/commands/appSettings/filterDownloadAppSettings.ts

```diff
@@ -4,65 +4,48 @@
 import { ext } from "../../extensionVariables";
 import { localize } from "../../localize";
 
 export async function filterDownloadAppSettings(context: IActionContext, sourceSettings: { [key: string]: string }, destinationSettings: { [key: string]: string }, destinationSettingsToIgnore: string[], destinationName: string): Promise<void> {
-    let suppressPrompt: boolean = false;
-    let overwriteSetting: boolean = false;
-    let overwriteSettingsToIgnore: boolean = false;
 
     const addedKeys: string[] = [];
     const updatedKeys: string[] = [];
     const userIgnoredKeys: string[] = [];
     const matchingKeys: string[] = [];
-    const securitySettingsIgnored: string[] = [];
+
     const listOfSettingsToIgnore: string[] = ["AzureWebJobsStorage", "WEBSITE_CONTENTAZUREFILECONNECTIONSTRING", "WEBSITE_CONTENTSHARE"];
+    const options: vscode.QuickPickItem[] = [];
+    destinationSettingsToIgnore.length = 0;
 
-    const yesToAll: vscode.MessageItem = { title: localize('yesToAll', 'Yes to all') };
-    const noToAll: vscode.MessageItem = { title: localize('noToAll', 'No to all') };
+    for (const element of Object.keys(sourceSettings)) {
+        options.push({
+            label: element,
+            picked: !listOfSettingsToIgnore.includes(element)
+        });
+    }
 
-    destinationSettingsToIgnore.length = 0;
+    const result = await context.ui.showQuickPick(options, { placeHolder: 'Select the app settings you would like to download:', canPickMany: true });
+    const userChosenSettings: string[] = result ? result.map(item => item.label) : [];
 
     for (const key of Object.keys(sourceSettings)) {
-        if (listOfSettingsToIgnore.includes(key)) {
-            if (!suppressPrompt) {
-                const message: string = localize('overwriteSettingsToIgnore', 'Setting "{0}" contains secrets. Download?', key);
-                const result: vscode.MessageItem = await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.yes, yesToAll, DialogResponses.no, noToAll);
-                overwriteSettingsToIgnore = result === DialogResponses.yes || result === yesToAll;
-                suppressPrompt = result === yesToAll || result === noToAll;
-            }
-            if (overwriteSettingsToIgnore) {
-                updatedKeys.push(key);
-                destinationSettings[key] = sourceSettings[key];
-            } else {
-                securitySettingsIgnored.push(key);
-                destinationSettings[key] = "_REDACTED_";
-                destinationSettingsToIgnore.push(key);
-            }
-        }
-        else {
-            if (destinationSettings[key] === undefined) { // Have an explicit check for undefined as empty settings should not pass the condition
+        if (userChosenSettings.includes(key)) {
+            if (destinationSettings[key] === undefined) { // Explicit check for undefined as destinationSettings[key] could be empty but valid
```

Comment:
Nit: Just my preference for comments, to have them on the line above instead of at the end of the line

Reply:
sure